src/main/scala/com/gu/subscriptions/cas/directives/ProxyDirective.scala

```diff
@@ -26,29 +26,39 @@ trait ProxyDirective extends Directives with ErrorRoute {
   implicit val actorSystem: ActorSystem
   lazy val subscriptionService: SubscriptionService = ZuoraSubscriptionService
 
-  lazy val casRoute = {
+  lazy val proxyHost = Configuration.proxyHost
+  lazy val proxyPort = Configuration.proxyPort
+  lazy val proxyScheme = Configuration.proxyScheme
+
+  def proxyRequest(in: HttpRequest): HttpRequest =
+    in.copy(
+      uri = in.uri.withHost(proxyHost).withPort(proxyPort).withScheme(proxyScheme),
+      headers = in.headers.map { header =>
```

Comment:
maybe here we could pattern match using the [Host](http://spray.io/documentation/1.1-SNAPSHOT/api/index.html#spray.http.HttpHeaders$$Host) case class

    headers.map { 
       case Host(_, _) => Host(proxyHost)
       case header => header
    }